toolchain/driver/clang_runner.cpp

```diff
@@ -42,9 +41,11 @@ auto clang_main(int Argc, char** Argv, const llvm::ToolContext& ToolContext)
 namespace Carbon {
 
 ClangRunner::ClangRunner(const InstallPaths* install_paths,
-                         llvm::StringRef target, llvm::raw_ostream* vlog_stream)
+                         llvm::StringRef target, llvm::vfs::FileSystem* fs,
+                         llvm::raw_ostream* vlog_stream)
     : installation_(install_paths),
       target_(target),
+      fs_(fs),
```

Comment:
We don't use the driver's VFS in other parts, and I'm concerned that an incomplete use will yield inconsistent results. Had you considered using the real filesystem here?

Reply:
The change causing this is allowing users to disable warnings on a per-file basis. Carbon would want to honour the same when building the same C++ code with the same paths.

Reply:
@chandlerc Should we use the real file system for everything or the vfs from the `driver_env` for everything?

Reply:
To be sure, I really think for the update PR you might be best off just maintaining current filesystem behavior, rather than changing it. However, if I'm in the minority here, note that `clang::driver::Driver` takes a `IntrusiveRefCntPtr`, so you'll want to change the types you're using for storage on `ClangRunner` and probably also `DriverEnv` and the `FileTest` infrastructure.

Reply:
Agreed that for an LLVM update PR, the smaller change is more appropriate, and any other changes can be done as a follow-up if desired. Updated.

Reply:
(agreed about not shuffling anything in the update PR)

For any follow-up change: I think we'll want to pass down the `driver_env` VFS -- basically, when we reach Clang functionality from the driver, we want Clang to see a consistent world as the non-Clang parts of the driver from a VFS perspective.